src/tasks/getBulkUploadLogDetailsFromError.ts

```diff
@@ -0,0 +1,59 @@
+import type { BulkUploadLogDetails } from '../types';
+
+export const CYCLE_ERROR_MESSAGE =
+	'A cycle was detected in the error cause chain.';
+
+/**
+ * Convert a thrown error into a serializable object with a `message` string.
+ * @param err A thrown error, may be Error or object.
+ * @param previouslySeenErrors A Set of previously seen errors to break cycles.
+ * @throws TypeError when JSON.stringify throws a (non-cause) circularity error.
+ * @returns A serializable object with at least a `message` property.
+ */
+export const getBulkUploadLogDetailsFromError = (
+	err: unknown,
+	previouslySeenErrors: Set<unknown> = new Set(),
+): BulkUploadLogDetails => {
+	if (previouslySeenErrors.has(err)) return { message: CYCLE_ERROR_MESSAGE };
+	previouslySeenErrors.add(err);
+	let objectFromError: {
```

Comment:
Rather than defining a mutable object here what do you think of simply having return statements that are explicitly crafted within each type-guard-block?

This would mean, for instance, something like:

```
	if (err instanceof Error) {
	  const cause = (err.cause !== undefined && err.cause !== null) ? getBulkUploadLogDetailsFromError(
				err.cause,
				previouslySeenErrors,
			) : undefined;
	  return {
			message: err.message,
			name: err.name,
			cause,
		};
	}
	```